### PR TITLE
Added infrastructure for custom groups testing

### PR DESCRIPTION
**Pull request by SergioBertolinSG**

### CODE REVIEW

**Comment by PVince81** [comment]
Looks fine 👍 

I'll add a Makefile later

**Comment by SergioBertolinSG** [comment]
Problem with changing autoloading:      

Adding two elements with hyphens it fails:

- '': %paths.base%/../../../../../build/integration/features/bootstrap
- '': %paths.base%/../features/bootstrap

Result:
```php
[Symfony\Component\Config\Definition\Exception\InvalidTypeException]          
  Invalid type for path "testwork.autoload.0". Expected scalar, but got array.
```

Without hyphens it loads only the content from the first path, not the second.

We need to figure out how to autoload contexts from both. 

Perhaps using profiles?
Perhaps using composer's PSR-4 autoload mechanism?

**Comment by SergioBertolinSG** [comment]
This https://github.com/owncloud/core/pull/27104 has to be merged before travis can pass.

File: .travis.yml [hunk 1]
```diff
@@ -28,16 +28,19 @@ before_install:
...
+  # Run integration tests
+  - sh -c "if [ '$INTEGRATION' = '1' ]; then cd tests/integration; ./run.sh; fi"
+
   # Test the app
-  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '1' ]; then make test-codecheck; fi"
-  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '1' ]; then make test-codecheck; fi"
-  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '2' ]; then make test-codecheck-deprecations; fi"
+  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '1' -a '$INTEGRATION' != '1' ]; then make test-codecheck; fi"
+  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '1' -a '$INTEGRATION' != '1' ]; then make test-codecheck; fi"
+  - sh -c "if [ '$JSTESTS' != '1' -a '$CODECHECK' = '2' -a '$INTEGRATION' != '1' ]; then make test-codecheck-deprecations; fi"
```
Comment (PVince81) [inline .travis.yml]: It starts to feel like we should have only a single variable "TEST_WHATTOTEST=integration" or other values.

Anyway, something for another time.

**Comment by PVince81** [comment]
Great start 👍

**Pull request merged**